### PR TITLE
demo(datepicker): Ammended datepicker adapter example to include cust…

### PR DESCRIPTION
**Pull request by gpolychronisAmadeus**
…om adapter and parser/formatter

On current adapter example added custom implementations for both adapter and parser/formatter. This commit might help users to:
- Avoid confusion (use one service instead of the other one).
- Write their own implementations, as they have templates for both services.

@maxokorokov 
Please review

### CODE REVIEW

File: demo/src/app/components/datepicker/demos/adapter/datepicker-adapter.ts [hunk 1]
```diff
@@ -1,20 +1,91 @@
 import {Component, Injectable} from '@angular/core';
-import {NgbDateAdapter, NgbDateStruct, NgbDateNativeAdapter} from '@ng-bootstrap/ng-bootstrap';
+import {
+  NgbCalendar,
+  NgbDateAdapter,
+  NgbDateStruct,
+  NgbDateNativeAdapter,
```
Comment (peterblazejewicz) [inline demo/src/app/components/datepicker/demos/adapter/datepicker-adapter.ts]: unused import

File: demo/src/app/components/datepicker/demos/adapter/datepicker-adapter.html [hunk 2]
```diff
@@ -1,6 +1,10 @@
-<p>These datepickers use custom Date adapter that lets you use your own model implementation.
-In this example we are converting from and to a JS native Date object</p>
-
+<p>These datepickers use custom date adapter and formatter.
+The custom NgbDateAdapter lets you use your own model i.e. the structure of ngModel.
```
Comment (peterblazejewicz) [inline demo/src/app/components/datepicker/demos/adapter/datepicker-adapter.html]: wrap with `<code>NgbDateAdapter</code>`, so it renders with visual clue <code>NgbDateAdapter</code>

File: demo/src/app/components/datepicker/demos/adapter/datepicker-adapter.html [hunk 3]
```diff
@@ -1,6 +1,10 @@
...
+The custom NgbDateParserFormatter lets you use your own formatter or parser i.e. the date format in input field.
```
Comment (peterblazejewicz) [inline demo/src/app/components/datepicker/demos/adapter/datepicker-adapter.html]: ditto

File: demo/src/app/components/datepicker/demos/adapter/datepicker-adapter.html [hunk 4]
```diff
@@ -1,6 +1,10 @@
...
+<br/>
+<br/>
+Notice that there are <strong>two</strong> different services: the NgbDateAdapter and the NgbDateParserFormatter.</p>
```
Comment (peterblazejewicz) [inline demo/src/app/components/datepicker/demos/adapter/datepicker-adapter.html]: ditto (<code> tag)

File: demo/src/app/components/datepicker/overview/datepicker-overview.component.html [hunk 5]
```diff
@@ -134,17 +134,17 @@ <h4>Handling the popup</h4>
   <h4>Adapters</h4>
 
   <p>
-    You can also tell datepicker to use the native javascript date adapter (bundled with ng-bootstrap) as in the
-    <a routerLink="../examples" fragment="adapter">custom date adapter example</a>. For now
-    the adapter works only for the form integration, so for instance <code>(ngModelChange)</code>
+    You can also tell datepicker to use the native javascript date adapter (bundled with ng-bootstrap).
```
Comment (peterblazejewicz) [inline demo/src/app/components/datepicker/overview/datepicker-overview.component.html]: I believe there was an error in copied statement, and the `JavaScript` is a proper name, so character case should be corrected

File: demo/src/app/components/datepicker/overview/datepicker-overview.component.html [hunk 6]
```diff
@@ -164,6 +164,12 @@ <h4>Input date parsing and formatting</h4>
     If the entered input value is invalid, the form model will contain the entered text.
   </p>
 
+  <p>
+    You can also create your own parser/formatter if necessary by extending and implementing the
```
Comment (peterblazejewicz) [inline demo/src/app/components/datepicker/overview/datepicker-overview.component.html]: `/` means `or` in plain English, can we amend this?

**Comment by gpolychronisAmadeus** [comment]
@peterblazejewicz 
Pull Request has been updated. Please review.
